www/docs/en/dev/platform_pinning/index.md

```diff
@@ -24,11 +24,11 @@ description: Cordova CLI pinned platform versions.
 
 ## Platform Pinning
 
-Cordova CLI pins the supported platform version in conjunction with the version of CLI that is being used.
+Cordova CLI manages a list of pinned Apache Cordova maintained platforms. When there is a major release of Cordova CLI, the platform's pinned version is updated to the latest major released version. Typically, the updating of this pin is made only on a major release of CLI. This is because the pinned versions are pinned with a caret `^` allowing CLI to continue to fetch new minor and patch releases of any given pinned platform.
```

Comment:
I will leave that up to the developer to decide when they do the release.

In most cases when we release a major CLI we had already released a major of each platform. There is typically no `.1` or `.2` or what ever at that time unless someone really screwed up a release or introduce a major break/security bug before CLI was released. Then yes the developer can start on that version...

If for any reason a platform was not released, the developer could also decided at that time to update to the last release but I think it would look a bit ugly... Also people may feel at that point that past minors/patches are no longer consider supported when bumped. It more like a range of what works and currently supported with the know major CLI.

Since we have the `^` it will always use latest minor/patch.

Maybe it can be argued, if it works, to just drop the `^` and change the entire format to `8.x` then it is always major. But this is not something to discuss in a PR about what we have now for docs.

Anyways there is many up to developer/releaser information and use cases. All this can be documented else where like contribute. I think all this is not necessary for end user to read. This generally outlines that we pin, when pins are updated, what would potential fetched.

Reply:
I think you misunderstood.

As example right now Cordova iOS is 5.0.1, pinned to `^5.0.0` in lib. If we do a iOS 5.1.0 now, and then release a new CLI, if I understood what you wrote correctly, there would be no need to update the pinning of Cordova iOS because the major didn't change.

But I think it would actually be beneficial to change the pinning to `^5.1.0` - even though it wouldn't change the installed version. But it improves the output you see when running `cordova platform list` and during  `cordova platform add ...` before it knows the actual version - it would give the user a better picture of what verison they will get.

Reply:
If we want to provide a `user a better picture of what version they will get.` then we should be fetching a list of versions from npm and then filter it by our pinned version and then grab the latest. Then it would show `5.0.1`, `5.1.0` or even `5.1.2`... 

Right now in the current structure, if we want to provide that information, we would have to do a patch release of LIB and CLI for every platform release. Break the release schedule down for this:

* Prepare and Submit Platform Release Vote 1 day
* 48 hours (2 days) release & prepare lib release vote
* 48 hours (2 days) release lib & prepare CLI release vote
* release
(Total time on current structure: approx. around 1 week)

Reply:
Anyways, if there was a patch release to LIB and CLI, the developer can update the pinnings at that time if wanted.. There isn't anything blocking them from doing it when they want. Just that Major to Major is typical. Not a major platform in a minor or patch of lib/cli.

Reply:
This is not what I suggested.

You define here that pinning only uses major versions:

> the platform's pinned version is updated to the latest major released version.

This is unnecessarily restricting. Use the latest or at least latest minor to pin, and the existing codebase can give more useful output to users. (Pre Cordova 9 we already pinned the latest available version of a platform - so I don't understand why this new limitation has to be added here just because the current pinning list only contains .0.0 versions.)

```suggestion
Cordova CLI manages a list of pinned Apache Cordova maintained platforms. When there is a major release of Cordova CLI, the platform's pinned version is updated to the latest released version. Typically, the updating of this pin is made only on a major release of CLI. This is because the pinned versions are pinned with a caret `^` allowing CLI to continue to fetch new minor and patch releases of any given pinned platform.
```

Reply:
Going back to what I said before,

> When there is a major release of Cordova CLI, the platform's pinned version is updated to the latest major released version. 

Typically by the time we release Cordova CLI 10 or any major version, it typically also inlines with a major platform release.

```
  android ^9.0.0
  browser ^7.0.0
  electron ^2.0.0
  ios ^6.0.0
  osx ^6.0.0
  windows ^8.0.0
```

But I also said in the past comment 

> If for any reason a platform was not released, the developer could also decide at that time to update to the last release but I think it would look a bit ugly...

That means the developer can change it to `^5.1.0` or what ever version it currently at. I just don't document all possible developer choice cases.

I can change the wording to this

```
When there is a major release of Cordova CLI, the platform's pinned version is updated latest released version. 
```

But I know in almost 99% of the time its major.